### DISCUSSION TITLE
Pass login_hint to external IDPs

### DISCUSSION THREAD

**Discussion by DavideWalder** [zitadel/zitadel#4922] [Ideas]
Currently, when logging in with Azure, `prompt=select_account` is set, but not `login_hint`. Would it be possible to pass it so that users don't have to select their account twice (once in the Zitadel login page and once on their IDP's)?

I saw that `login_hint` is already implemented in https://github.com/zitadel/oidc/blob/b5da6ec29b7e601ece7c97b8c1f4334cba8acd2b/pkg/oidc/authorization.go#L79, I might have missed how to use it if there is already an option to do so.

**Comment by mffap** [comment]
You can pass a valid username in the `login_hint` in combination with `prompt=select_account` as part of you auth request. How you pass it, depends a bit on the library that you use. 

You should be able to try the configuration out in the [playground](https://zitadel.com/docs/apis/openidoauth/authrequest?prompt=select_account). Given the select_account and a username of your user you should be redirected without the account picker. You need to fill in the required parameters first.

Hope I understood your question correctly.

**Reply by DavideWalder** [reply]
If I understand your suggestion correctly, you are referring to the Zitadel login (that is the authorization request sent from a webapp). I was referring to the autorization request sent to the external IDP, example:

- First time user enters its login name/email address
- Domain discovery selects the organization and directly sends to the IDP login
- The user still has to enter its login name (or select its account), since the login hint is not passed from Zitadel to the IDP

**Reply by mffap** [reply]
Ok, thanks for the clarification. We would probably need to pass on parameters dynamically to IDPs, f.e. something similar like [auth0](https://auth0.com/docs/authenticate/identity-providers/pass-parameters-to-idps#dynamic-parameters) does. 

@livio-a do we we support something similar currently?
